packages/squiggle-lang/src/rescript/Distributions/SampleSetDist/KdeLibrary.js

```diff
@@ -1,33 +1,84 @@
-const pdfast = require("pdfast");
 const filter = require("lodash/filter");
 const isFinite = require("lodash/isFinite");
 
-const samplesToContinuousPdf = (
-  samples,
-  size,
-  width,
-  min = false,
-  max = false
-) => {
-  let _samples = filter(samples, isFinite);
-  if (isFinite(min)) {
-    _samples = filter(_samples, (r) => r > min);
-  }
-  if (isFinite(max)) {
-    _samples = filter(_samples, (r) => r < max);
-  }
-
-  // The pdf that's created from this function is not a pdf but a pmf. y values
-  // being probability mass and not density.
-  // This is awkward, because our code assumes later that y is a density
-  let pdf = pdfast.create(_samples, { size, width });
-
-  // To convert this to a density, we need to find the step size. This is kept
-  // constant for all y values
-  let stepSize = pdf[1].x - pdf[0].x;
-
-  // We then adjust the y values to density
-  return { xs: pdf.map((r) => r.x), ys: pdf.map((r) => r.y / stepSize) };
+// Convert samples to x-y pairs for a PDF
+// Uses kernel density estimation (KDE) with a triangular kernel
+// samples: Must be sorted!
+// size:    Number of points in output, >= 4
+// xWidth:  Width of the kernel in x axis units
+// weight:  Probability mass for each point
```

Comment:
I appreciate the comments, but think it could use a fair bit more. This is complex stuff. First, I'd want some high-level explanation on top of what the algorithm is doing.

Reply:
We may need to have a discussion about what's expected regarding comments. This is very surprising to me given that the rest of the codebase has few comments, and overall I'm not so sure what I'm supposed to be doing.

As plain text, comments aren't a great medium for high-level explanation. If we want to do much of this I'd prefer to write something public-facing in the style of [this page](https://github.com/mlochbaum/CrinGraph/blob/master/Documentation.md) that can include links and equations. For the KDE in particular, my high-level view is really just that it implements KDE with a triangular kernel. Explaining what this means is much better covered by public resources than I can do.

Reply:
A discussion sounds good to me.

I think for at least Slava and myself, this sort of code is particularly terse and optimized much more for performance than readability. (Most of my code is typically optimized heavily for readability). 

We've had bits like this before, but usually they were really isolated. (Sam did 2-4 days work, then would go on to do lots of more simple stuff).

In your case it seems like you're going to do lots of performance-optimized stuff, and for a long time, so I'm much more concerned, especially early on.